### ISSUE TITLE
Intermittent crash on net module PUT request

### ISSUE THREAD

**Issue by kwonoj** [electron/electron#21151]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
7.1.1
* **Operating System:**
  * 
* **Last Known Working Electron version:**
  * 6.x

Windows 10 / Mac Os catalina

### Actual Behavior
DCHECK failure

```
[9628:1115/193639.015:FATAL:atom_api_url_request_ns.cc(467)] Check failed: producer_.
Backtrace:
        base::debug::CollectStackTrace [0x00007FF7A1C5B152+18] (o:\base\debug\stack_trace_win.cc:284)
        base::debug::StackTrace::StackTrace [0x00007FF7A1C56F72+18] (o:\base\debug\stack_trace.cc:203)
        logging::LogMessage::~LogMessage [0x00007FF7A1C75863+115] (o:\base\logging.cc:630)
        electron::api::URLRequestNS::DoWrite [0x00007FF79F0C0E83+99] (o:\electron\shell\browser\api\atom_api_url_request_ns.cc:468)
        electron::api::URLRequestNS::Write [0x00007FF79F0C0271+881] (o:\electron\shell\browser\api\atom_api_url_request_ns.cc:276)
        mate::internal::Invoker<mate::internal::IndicesHolder<0,1,2>,electron::api::URLRequestNS *,v8::Local<v8::Value>,bool>::DispatchToCallback<bool> [0x00007FF79F0C3E7A+102] (o:\electron\native_mate\native_mate\function_template.h:200)
        mate::internal::Dispatcher<bool (electron::api::URLRequestNS *, v8::Local<v8::Value>, bool)>::DispatchToCallback [0x00007FF79F0C3DAF+159] (o:\electron\native_mate\native_mate\function_template.h:244)
        v8::internal::FunctionCallbackArguments::Call [0x00007FF7A082CF98+1080] (o:\v8\src\api\api-arguments-inl.h:159)
        v8::internal::`anonymous namespace'::HandleApiCallHelper<0> [0x00007FF7A07C1D9A+826] (o:\v8\src\builtins\builtins-api.cc:113)
        v8::internal::Builtin_Impl_HandleApiCall [0x00007FF7A07C03C5+453] (o:\v8\src\builtins\builtins-api.cc:141)
        v8::internal::Builtin_HandleApiCall [0x00007FF7A07BFEC0+160] (o:\v8\src\builtins\builtins-api.cc:129)
        Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit [0x00007FF7A15B2F44+68]
        Builtins_InterpreterEntryTrampoline [0x00007FF7A12E159A+954]
        Builtins_InterpreterEntryTrampoline [0x00007FF7A12E159A+954]
        Builtins_ArgumentsAdaptorTrampoline [0x00007FF7A12C5DF9+185]
        Builtins_InterpreterEntryTrampoline [0x00007FF7A12E159A+954]
        Builtins_InterpreterEntryTrampoline [0x00007FF7A12E159A+954]
        (No symbol) [0x000003E6BB7EF04C]
Task trace:
Backtrace:
        electron::NodeBindings::WakeupMainThread [0x00007FF79F178E97+235] (o:\electron\shell\common\node_bindings.cc:427)
```

### To Reproduce

@nornagon and I occasionally hit similar exception recently but so far not able to create isolated repro case can share here. So far this happens PUT + chunkedEncoding.

**Comment by kwonoj** [comment]
Above stack is from my dev's build on wnidows, prod stacks are like below:

(from mac)
```
mojo::DataPipeProducer::InitializeNewRequest(base::OnceCallback<void (unsigned int)>) (in Electron Framework) (data_pipe_producer.cc:190)
mojo::DataPipeProducer::Write(std::__1::unique_ptr<mojo::DataPipeProducer::DataSource, std::__1::default_delete<mojo::DataPipeProducer::DataSource> >, base::OnceCallback<void (unsigned int)>) (in Electron Framework) (data_pipe_producer.cc:178)
electron::api::URLRequestNS::DoWrite() (in Electron Framework) (atom_api_url_request_ns.cc:469)
electron::api::URLRequestNS::Write(v8::Local<v8::Value>, bool) (in Electron Framework) (atom_api_url_request_ns.cc:276)
mate::internal::Dispatcher<bool (electron::api::URLRequestNS*, v8::Local<v8::Value>, bool)>::DispatchToCallback(v8::FunctionCallbackInfo<v8::Value> const&) (in Electron Framework) (function_template.h:243)
v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) (in Electron Framework) (api-arguments-inl.h:159)
v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) (in Electron Framework) (builtins-api.cc:113)
v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) (in Electron Framework) (builtins-api.cc:0)
Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit (in Electron Framework) + 57
Builtins_InterpreterEntryTrampoline (in Electron Framework) + 673
Builtins_InterpreterEntryTrampoline (in Electron Framework) + 673
Builtins_ArgumentsAdaptorTrampoline (in Electron Framework) + 185
Builtins_InterpreterEntryTrampoline (in Electron Framework) + 673
Builtins_InterpreterEntryTrampoline (in Electron Framework) + 673
Builtins_InterpreterEntryTrampoline (in Electron Framework) + 673
Builtins_InterpreterEntryTrampoline (in Electron Framework) + 673
Builtins_InterpreterEntryTrampoline (in Electron Framework) + 673
Builtins_JSEntryTrampoline (in Electron Framework) + 90
Builtins_JSEntry (in Electron Framework) + 120
v8::internal::(anonymous namespace)::Invoke(v8::internal::Isolate*, v8::internal::(anonymous namespace)::InvokeParams const&) (in Electron Framework) (execution.cc:0)
```

**Comment by nornagon** [comment]
Seems similar to #20773. Should we dup this to that issue?

**Comment by kwonoj** [comment]
Stack seems hitting very similar execution path, but can't say for sure if #20773 uses `chunkedEncoding` as well. If so, I think we can dupe. (If not, means there's other race condition than this)

**Comment by nornagon** [comment]
ah, yeah, seems like #20773 is related to WebRequest so probably different.

**Comment by nornagon** [comment]
This was probably fixed by #21244. Please reopen if you're still seeing this!
